src/Kriti/Parser.hs

```diff
@@ -249,6 +286,26 @@ parserIff = do
   pos2 <- fromSourcePos <$> P.getSourcePos
   pure $ Iff (pos1, Just pos2) p t1 t2
 
+registerParseErrorBundle :: P.MonadParsec e s m => P.ParseErrorBundle s e -> m ()
+registerParseErrorBundle (P.ParseErrorBundle errs _) =
+  traverse_ P.registerParseError errs
+
+parserStringInterp :: Parser ValueExt
+parserStringInterp = do
+  pos1 <- fromSourcePos <$> P.getSourcePos
+  tem <- stringTem
+  pos2 <- fromSourcePos <$> P.getSourcePos
+  -- TODO: Recursively parsing like this will result in a junk
+  -- `SourcePosition`
+  let p = first ParseError . P.runParser parsePath mempty
+  x <- traverse (pure . traverse p) tem
```

Comment:
Here we recursively parse the lexed template expression and then register any errors with our main parsing process. This results in bad SourcePositions but I couldn't think of any other ways to get the job done.